pdf_view.php

```diff
@@ -50,39 +61,40 @@
 	$isInternalRequest = true;
 	return ( require $type . '_view.php' );
 })( $module, $listReports[ $reportData['source'] ]['type'] );
+$listVars = [];
 
 
 // Get the HTML for the PDF.
 $inputHTML = $reportData['pdf'];
 
 
 // Parse conditional and loop instructions.
-$inputHTML = preg_split( '/(<\\?(?:end|loop|if(\\((?:[^)(\'"]+|\'[^\']*\'|"[^"]*"|(?2))*+\\)))>)/',
+$inputHTML = preg_split( '/(<\\?(?:end|loop|(?:if|setvar\\[([^\\[\\]]+)\\])' .
```

Comment:
I must admit I can't understand this bit - could you expand on the comment above to explain what is going on and why?